infor-design/enterprise
# 740/2507 - Add an option to clear the field

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
Added an ability to clear the field in lookup component.  It supports both single and multiple values.

Make `clearable: true`, it will show `x` button in the field (It inherits in searchfield component).

Also adds `apply` button in `selectable: single`

**Related github/jira issue (required)**:
Closes
- #740
- #2507

**Steps necessary to review your pull request (required)**:
- Pull this branch, build, and run the app
- Navigate to http://localhost:4000/components/lookup/example-clearable.html

1.) Single Select

Scenario 1
- Click the first field `Products (Single)` icon to trigger the lookup modal
- Notice, it has already an `apply` button. Select any item
- That item should now populated in the field and that will show the `x` icon button
- Open again the lookup modal
- Deselect the item and hit `apply` button
- Should clear the in the field

Scenario 2
- Open and select an item
- It will show it again in the field
- Click `x` button. Should clear the field
- Try to type anything, should trigger the `x` button and can clear that too
- Test this on uplift

2.) Multiple Select
- Open the lookup modal
- Select all and hit `apply` button
- `x` button should show, text should truncated, and it should not overlap on those icons
- Hit `x` button, should clear the field
- Test this also on uplift

**Included in this Pull Request**:
